### PR TITLE
swapRanges: Remove redundant constraint tests

### PR DESCRIPTION
**Pull request by ntrel**
Also fix related `hasSwappableElements `docs.

`hasSwappableElements` already tests for `isInputRange`.

### CODE REVIEW

File: std/algorithm/mutation.d [hunk 1]
```diff
@@ -2728,9 +2728,8 @@ Returns:
 */
 Tuple!(InputRange1, InputRange2)
 swapRanges(InputRange1, InputRange2)(InputRange1 r1, InputRange2 r2)
-    if (isInputRange!(InputRange1) && isInputRange!(InputRange2)
-            && hasSwappableElements!(InputRange1) && hasSwappableElements!(InputRange2)
-            && is(ElementType!(InputRange1) == ElementType!(InputRange2)))
+    if (hasSwappableElements!(InputRange1) && hasSwappableElements!(InputRange2)
+        && is(ElementType!(InputRange1) == ElementType!(InputRange2)))
```
Comment (wilzbach) [inline std/algorithm/mutation.d]: How about removing the parentheses all-together? They are probably an relict from a time where `A!B` wasn't allowed in D?

Comment (ntrel) [inline std/algorithm/mutation.d]: Sure, will do.

**Comment by JackStouffer** [comment]
Auto-merge toggled on

**Comment by wilzbach** [comment]
Auto-merge toggled off

**Comment by wilzbach** [comment]
> Auto-merge toggled off

Oh seems we did it at the same time :P

**Pull request merged**